code/modules/mob/new_player/preferences_setup.dm

```diff
@@ -257,15 +257,21 @@ datum/preferences
 				preview_icon.Blend(markings_s, ICON_OVERLAY)
 
 
-		var/icon/eyes_s = new/icon("icon" = 'icons/mob/human_face.dmi', "icon_state" = current_species ? current_species.eyes : "eyes_s")
-		eyes_s.Blend(rgb(r_eyes, g_eyes, b_eyes), ICON_ADD)
+		var/icon/face_s = new/icon()
+		if(!(current_species.bodyflags & NO_EYES))
+			var/icon/eyes_s = new/icon("icon" = 'icons/mob/human_face.dmi', "icon_state" = current_species ? current_species.eyes : "eyes_s")
+			eyes_s.Blend(rgb(r_eyes, g_eyes, b_eyes), ICON_ADD)
+			face_s.Blend(eyes_s, ICON_OVERLAY)
 
 
 		var/datum/sprite_accessory/hair_style = hair_styles_list[h_style]
 		if(hair_style)
 			var/icon/hair_s = new/icon("icon" = hair_style.icon, "icon_state" = "[hair_style.icon_state]_s")
-			hair_s.Blend(rgb(r_hair, g_hair, b_hair), ICON_ADD)
-			eyes_s.Blend(hair_s, ICON_OVERLAY)
+			if(current_species.name == "Slime People") // whee I am part of the problem
+				hair_s.Blend(rgb(r_skin, g_skin, b_skin,160), ICON_ADD)

```

Comment:
Should probably put a space before that 160 just for consistency (see update_icons.dm lines 466 and 503)